v2/contribute/decisions/accountlinking/0005-allowing-email-verification-in-post-login-account-linking.mdx

```diff
@@ -0,0 +1,66 @@
+        - Attacker tries to create an account with email B which doesn't belong to them will send a verification mail to actual user.
+        - Actual user clicks the verification link and the email password account gets linked to the attacker's account.
+        - If later the actual user signs up with email B via some other flow, those accounts will be linked to the attacker's account
+    - Scenario 2:
+        - There already exists an account with email B which is not associated with a primaryUserId because the actual user has not verified the account
+        - Attacker creates an account with email A, which belongs to them and a primaryUserId gets created
+        - Now attacker initiate post login account linking flow where the session exists with email A and account to be linked is B (which already exists but without primaryUserId)
+        - If email verification mail is sent to the actual user, their account will then linked to the attacker's account
+- But those attack vectors could be executed anyway, regardless of this API not allowing account linking, examples:
+    - Counter to Scenario 1:
+        - Attacker creates an account with email B, which doesn't belongs to them and during sign-up verification mail is sent
+        - Actual user clicks the verification link and the email password account gets verified, thus creating a primaryUserId. The attacker knows the login info here cause they were the one who created the account in the first place
+        - Now attacker initiate post login account linking flow where the session exists with email B and account to be linked is A which belongs to them
+        - Thus, the state that we tried to prevent by disallowing email verification flow during post login account linking flow can still be achieved via normal sign-up flow
+        - Also, the actual user clicking on the verification link without realising whether they have a account for a given service is fault on user's end. This can happen for any service existing today and disallowing email verification during post login account linking flow is not the right solution
+    - Counter to Scenario 2:
+        - When the attacker initiates post login account linking flow where the session exists with email A and account to be linked is B (which already exists but without primaryUserId), they also needs to know the login info for account with email B. Also, the account was not created by them. So if they know the login info here, it could be be a user mistake where they would have disclosed there login info somehow and the attacker got the hold of it. This has nothing to do with disallowing email verification flow and can happen regardless.
+
+### Further information
+
+- We decided that if existing email is not verified && shouldRequireVerification is true, we will initiate the email verification flow without keeping track of the new account to be linked - because this is just easier to implement.
```

Comment:
```suggestion
- We decided that in the post login account linking API, if the account belonging to the session is not verified && shouldRequireVerification is true, we will initiate the email verification flow without keeping track of the new account to be linked - because this is just easier to implement.
```